ui/app/components/job-dispatch.js

```diff
@@ -0,0 +1,138 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+import { A } from '@ember/array';
+import { task } from 'ember-concurrency';
+import { noCase } from 'no-case';
+import { titleCase } from 'title-case';
+import messageFromAdapterError from 'nomad-ui/utils/message-from-adapter-error';
+
+class MetaField {
+  @tracked value;
+  @tracked error;
+
+  name;
+  required;
+  title;
+
+  constructor(meta) {
+    this.name = meta.name;
+    this.required = meta.required;
+    this.title = meta.title;
+    this.value = meta.value;
+    this.error = meta.error;
+  }
+
+  validate() {
+    this.error = '';
+
+    if (this.required && !this.value) {
+      this.error = `Missing required meta parameter "${this.name}".`;
+    }
+  }
+}
+
+export default class JobDispatch extends Component {
+  @service router;
+  @service config;
+
+  @tracked metaFields = [];
+  @tracked payload = '';
+  @tracked payloadHasError = false;
+
+  errors = A([]);
+
+  constructor() {
+    super(...arguments);
+
+    // Helper for mapping the params into a useable form.
+    const mapper = (values, required) =>
+      values.map(
+        x =>
+          new MetaField({
+            name: x,
+            required: required,
+            title: titleCase(noCase(x)),
+            value: this.args.job.meta ? this.args.job.meta[x] : '',
+          })
+      );
+
+    // Fetch the different types of parameters.
+    const required = mapper(this.args.job.parameterizedDetails.MetaRequired || [], true);
```

Comment:
No change needed, JS ES6 Knowledge Transfer Comment:

We're using `||` in the event that we have `null` or `undefined` here?

You should checkout the nullish coalescing operator `??` which specifically handles the cases of `null` and `undefined` but will return the left hand operand if its a falsy value like `0` or `''`.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator

Reply:
Ah good to know, thanks! In this case it doesn't seem to matter, since falsey values are not expected for our `Meta*` attributes.

Reply:
I actually changed the function signature to:
```js
const mapper = (values = [], required) {
  ...
}
```

Seems more readable and reliable. Thanks for making double check 😄